doc/Command-Reference.md

```diff
@@ -4119,6 +4119,18 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#Warm-R
 
 ## Warm Restart
 
+Besides device level warm reboot, SoNIC also provides docker based warm restart. This feature is currently supported by following dockers: BGP, teamD,  and SWSS. A user can manage to restart a particular docker, with no interruption on packet forwarding and no effect on other services. This helps to reduce operational costs as well as development efforts. For example, to fix a bug in BGP routing stack, only the BGP docker image needs to be built, upgraded and tested.
+
+To achieve uninterrupted packet forwarding during the restarting stage and database reconciliation at the post restarting stage, warm reboot enabled dockers with adjacency state machine facilitate standardized protocols. For example, a BGP restarting must have BGP graceful restart enabled and its BGP neighbors must be graceful restart helper capable, which are specified in [IETF RFC4724](https://tools.ietf.org/html/rfc4724). Before warm restart BGP docker, the following BGP commands should be configured:
+
+  bgp graceful-restart
+  bgp graceful-restart preserve-fw-state
+
+It should be aware that during a warm reboot, some fast convergence features, such as BFD, fast Hello, must be disabled. Otherwise, the fast timers would affect warm restart and cause packet drop.
```

Comment:
Can you add guidance how end user should check / disable these features?

Reply:
Added, with more configuration